causalpy/pymc_models.py

```diff
@@ -23,29 +43,63 @@ def __init__(self, sample_kwargs: Optional[Dict[str, Any]] = None):
         self.sample_kwargs = sample_kwargs if sample_kwargs is not None else {}
 
     def build_model(self, X, y, coords) -> None:
-        """Build the model.
+        """Build the model, must be implemented by subclass.
 
         Example
         -------
+        >>> import pymc as pm
+        >>> from causalpy.pymc_models import ModelBuilder
         >>> class CausalPyModel(ModelBuilder):
-        >>>    def build_model(self, X, y):
-        >>>        with self:
-        >>>            X_ = pm.MutableData(name="X", value=X)
-        >>>            y_ = pm.MutableData(name="y", value=y)
-        >>>            beta = pm.Normal("beta", mu=0, sigma=1, shape=X_.shape[1])
-        >>>            sigma = pm.HalfNormal("sigma", sigma=1)
-        >>>            mu = pm.Deterministic("mu", pm.math.dot(X_, beta))
-        >>>            pm.Normal("y_hat", mu=mu, sigma=sigma, observed=y_)
+        ...     def build_model(self, X, y):
+        ...         with self:
+        ...             X_ = pm.MutableData(name="X", value=X)
+        ...             y_ = pm.MutableData(name="y", value=y)
+        ...             beta = pm.Normal("beta", mu=0, sigma=1, shape=X_.shape[1])
+        ...             sigma = pm.HalfNormal("sigma", sigma=1)
+        ...             mu = pm.Deterministic("mu", pm.math.dot(X_, beta))
+        ...             pm.Normal("y_hat", mu=mu, sigma=sigma, observed=y_)
         """
         raise NotImplementedError("This method must be implemented by a subclass")
 
     def _data_setter(self, X) -> None:
+        """
+        Set data for the model.
+
+        This method is used internally to register new data for the model for
+        prediction.
+        """
         with self.model:
             pm.set_data({"X": X})
 
     def fit(self, X, y, coords: Optional[Dict[str, Any]] = None) -> None:
-        """Draw samples from posterior, prior predictive, and posterior predictive
-        distributions.
+        """Draw samples fromposterior, prior predictive, and posterior predictive
+        distributions, placing them in the model's idata attribute.
+
+        .. note::
+            Calls the build_model method
```

Comment:
Remove I think. Users don't need to know about the build_model method, that's not called by end-users. We should consider making this a private class, but that can be discussed elsewhere.